File: app/assets/javascripts/ResearchSubject.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const ResearchSubjectSchema = DataElementSchema({
  identifier: Any,
  status: Any,
  period: Any,
  study: Any,
  individual: Any,
  assignedArm: Any,
  actualArm: Any,
  consent: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'ResearchSubject' },

});

module.exports.ResearchSubjectSchema = ResearchSubjectSchema;
module.exports.ResearchSubject = mongoose.model('ResearchSubject', ResearchSubjectSchema);

File: app/models/qdm/research_subject.rb
module QDM
  # app/models/qdm/research_subject.rb
  class ResearchSubject < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :identifier
    field :status
    field :period
    field :study
    field :individual
    field :assignedArm
    field :actualArm
    field :consent
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
